### ISSUE TITLE
only detect one class

### ISSUE THREAD

**Issue by g1zzy316** [roboflow/rf-detr#210]
I defined only one class in my dataset and did fine-tuning. but on testing it works poorly and detects other classes, how do I adjust 
model = RFDETRBase(num_classes=2)
 model.train(
    dataset_dir="D:/pythonProject/cam_radar/dataset",  
    output_dir="D:/pythonProject/cam_radar/model",   
    
 
    epochs=10,
    batch_size=4,
    grad_accum_steps=4,
    lr=1e-4,
    lr_encoder=1.5e-4,

   
    resolution=672,   

    
    weight_decay=1e-4,
    use_ema=True,
    checkpoint_interval=10,
    tensorboard=True,

    early_stopping=True,
    early_stopping_patience=20,
    early_stopping_min_delta=0.001,
    early_stopping_use_ema=True,

 
    device="cuda",  
    gradient_checkpointing=False,  #

**Comment by Ahmed-Samir11** [comment]
You say you fine-tuned on one class, but your code shows num_classes = 2. Maybe that's a typo?

**Comment by g1zzy316** [comment]
Whatever I set num_classes to is forced to 90，this is log:
Loading pretrain weights
num_classes mismatch: pretrain weights has 90 classes, but your model has 2 classes
reinitializing detection head with 90 classes``

**Comment by Ahmed-Samir11** [comment]
@g1zzy316 The reason is that you didn't initialize the model you fine-tuned. To initialize its weights, you have to write:

`model = RFDETRBase(num_classes=2, pretrain_weights = "path/to/your/fine-tunedModel.pt")
`
Since you didn't initialize it, the initialized weights are of the base model, which is why it reverts to 90 classes even after specifying `num_classes = 2`.

**Comment by Ahmed-Samir11** [comment]
@g1zzy316 Your post says you already fine-tuned it. Can you check your codes? My example here works just fine and it tunes on a single class: 
```
def train_rfdetr():

    model = RFDETRBase(num_classes=1)
    
    # Custom callback for tracking metrics
    history = []
    def training_callback(data):
        history.append(data)
    model.callbacks["on_fit_epoch_end"].append(training_callback)

    model.train(
        dataset_dir="/kaggle/working/dataset",
        train_ann_file="/kaggle/working/dataset/train/_annotations.coco.json",
        val_ann_file="/kaggle/working/dataset/val/_annotations.coco.json",
        epochs=100,
        batch_size=4,  
        lr=0.00005,
        grad_accum_steps=4,
        checkpoint_interval=50,
        freeze_encoder=True,
        freeze_projection=True,
        freeze_batch_norm=True, 
        lr_vit_layer_decay=0.95,  
        lr_component_decay=0.85,
        warmup_epochs=3,  
        dropout=0.3, 
        #drop_path=0.1, 
        layer_norm=True,
        augmentation_strength=0.6, 
        ema_decay=0.999,
        early_stopping=True,
        early_stopping_patience=25
    )
    
    return model, history
model, history = train_rfdetr()
```

**Comment by g1zzy316** [comment]
I tried to fine tune it with your code and the issue still occurs. The logs are as follows: Loading pretrain weights
Class mismatch: pretrain weights have 90 classes and your model only has 2 classes
Reinitialize detection head with 90 classes
`from rfdetr import RFDETRBase
def train_rfdetr():

    model = RFDETRBase(num_classes=2)
    
    # Custom callback for tracking metrics
    history = []
    def training_callback(data):
        history.append(data)
    model.callbacks["on_fit_epoch_end"].append(training_callback)

    model.train(
        dataset_dir="cam_radar/dataset",
        train_ann_file="D:/pythonProject/cam_radar/dataset/train/_annotations.coco.json",
        val_ann_file="D:/pythonProject/cam_radar/dataset/valid",
        output_dir="output",
        epochs=50,
        batch_size=4,  
        lr=0.00005,
        grad_accum_steps=4,
        checkpoint_interval=50,
        freeze_encoder=True,
        freeze_projection=True,
        freeze_batch_norm=True, 
        lr_vit_layer_decay=0.95,  
        lr_component_decay=0.85,
        warmup_epochs=3,  
        dropout=0.3, 
        #drop_path=0.1, 
        layer_norm=True,
        augmentation_strength=0.6, 
        ema_decay=0.999,
        early_stopping=True,
        early_stopping_patience=25
    )
    
    return model, history
model, history = train_rfdetr()`

**Comment by g1zzy316** [comment]
I found what my problem was, thank you so much for your help!

**Comment by SilvaneUX** [comment]
hi, @g1zzy316 could you specify what your problem was and how do you fix it?

**Comment by Borda** [comment]
✅ This has been fixed in v1.6.0! Key changes:

- **Detection head reinitialization** no longer overwrites fine-tuned weights
- **`class_names`** now correctly reflect your dataset labels after training
- Misleading `num_classes` warnings were fixed in v1.4.2
- v1.6.0 also added composable Lightning training with proper `num_classes` handling

To upgrade: `pip install rfdetr>=1.6.1`

For the canonical fine-tuning guide, see #878. Closing as resolved — feel free to reopen if the issue persists on v1.6.1!
